opm/material/thermal/EclThconrLaw.hpp

```diff
@@ -56,12 +56,15 @@ class EclThconrLaw
                                           const FluidState& fluidState)
     {
         // THCONR + THCONSF approach.
-        Scalar lambdaRef = params.referenceTotalThermalConductivity();
-        Scalar alpha = params.dTotalThermalConductivity_dSg();
-
+        Scalar lambdaRef = params.referenceTotalThermalConductivity();        
         static constexpr int gasPhaseIdx = FluidSystem::gasPhaseIdx;
-        const Evaluation& Sg = Opm::decay<Evaluation>(fluidState.saturation(gasPhaseIdx));
-        return lambdaRef*(1.0 - alpha*Sg);
+        if(FluidSystem::phaseIsActive(gasPhaseIdx)){
```

Comment:
I'm not terribly happy with this code as it hard-codes phase names and it depends on a non-standard method in the FluidSystem (the former was also the case before). it would be nice at least the fluid system API situation would not be regressed: can you leave it as-is and just not compute the thermal conductivity for inactive phases in the intensive quantities?

Reply:
For me it seems lik this is called for all thermal runs and would make problems for cases where gas i not active. I think the change is needed. What is the non standard names?

Reply:
yes, the point was that the calling code should not need this quantity in the first place, i.e. arguably there is a small bug there!?

> What is the non standard names?

everything which is blackoil specific, i.e., everything which does not make sense if you go to "base thermodynamics". such stuff is e.g., the gas dissolution factor, the concept of "active" phases or even the names of phases or components. If you want to find out if black-oil specific stuff crept into your code, you can try to compile it with e.g. [`TwoPhaseImmiscibleFluidSystem`](https://github.com/OPM/opm-material/blob/master/opm/material/fluidsystems/TwoPhaseImmiscibleFluidSystem.hpp)